google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionImpl.java

```diff
@@ -78,6 +79,11 @@ static TransactionOptions createReadWriteTransactionOptions(Options options) {
     if (options.withOptimisticLock() == Boolean.TRUE) {
       readWrite.setReadLockMode(TransactionOptions.ReadWrite.ReadLockMode.OPTIMISTIC);
     }
+    if (previousTransactionId != null
```

Comment:
Updated prevTransactionId to have ByteString.Empty as default instead of null.
However, I have this null check in this if condition alone to ensure that we don't see NullPointerExceptions.